### PR TITLE
Babel 6 migration

### PR DESCRIPTION
**Pull request by pascalduez**
Unit tests <3

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
-  "optional": ["runtime"],
-  "auxiliaryComment": "istanbul ignore next"
+  "presets": ["latest"],
+  "plugins": [
+    ["transform-runtime", {
+      "polyfill": true,
+      "regenerator": true
+    }],
+    ["babel-plugin-transform-builtin-extend", {
+      "globals": ["Error"]
```
Comment (HugoGiraudel) [inline .babelrc]: Why do we need this?

**Comment by HugoGiraudel** [comment]
Looks good. :)

File: .babelrc (hunk 1 above)
Comment (pascalduez) [inline .babelrc]: Breaking change between Babel 5 and 6: https://github.com/babel/babel/issues/3083

We probably shouldn't to that apparently.  
https://github.com/SassDoc/sassdoc/blob/master/src/errors.js

**Comment by pascalduez** [comment]
I downgraded the ESLint parser for now, so that the tests pass. 

Next step: rework the linting!

**Comment by HugoGiraudel** [comment]
Any particular reason why code coverage decreased?

**Comment by pascalduez** [comment]
> Any particular reason why code coverage decreased?

Switch to `nyc`, and the use of that babel plugin for Errors.
I tried to investigate how we can improve things, sourcemaps'n stuff, but the difference is just to small to care of it.
Also we may switch to Jest at some point, which will make things easier.

**Comment by HugoGiraudel** [comment]
🎉

**Comment by pascalduez** [comment]
I'm trying to understand/update the dependencies warnings from the default-theme, and then I'll make a release. So that you will be able to merge the linting branch @HugoGiraudel ;)

**Pull request merged**